src/System.Private.ServiceModel/src/System/IdentityModel/Claims/X509CertificateClaimSet.cs

```diff
@@ -380,52 +379,108 @@ public IIdentity Identity
         // to figure out what the identifier, delimiter, and separator is by using a well-known extension
         private static class X509SubjectAlternativeNameConstants
         {
-            public const string Oid = "2.5.29.17"; 
-            public static readonly string Identifier;
-            public static readonly char Delimiter;
-            public static readonly string Separator;
+            public const string Oid = "2.5.29.17";
+
+            private static readonly string _identifier;
+            private static readonly char _delimiter;
+            private static readonly string _separator;
+
+            private static bool _successfullyInitialized = false;
+            private static Exception _initializationException;
+
+            public static string Identifier
+            {
+                get
+                {
+                    EnsureInitialized();
+                    return _identifier;
+                }
+            }
+
+            public static char Delimiter
+            {
+                get
+                {
+                    EnsureInitialized();
+                    return _delimiter;
+                }
+            }
+            public static string Separator
+            {
+                get
+                {
+                    EnsureInitialized();
+                    return _separator;
+                }
+            }
+
+            private static void EnsureInitialized()
+            {
+                if (!_successfullyInitialized)
+                {
+                    throw new FormatException(string.Format(
+                        "There was an error detecting the identifier, delimiter, and separator for X509CertificateClaims on this platform.{0}Detected values were: Identifier: '{1}'; Delimiter:'{2}'; Separator:'{3}'",
+                        Environment.NewLine,
+                        _identifier,
+                        _delimiter,
+                        _separator
+                    ), _initializationException);
+                }
+            }
 
             // static initializer runs only when one of the properties is accessed
             static X509SubjectAlternativeNameConstants()
             {
                 // Extracted a well-known X509Extension
-                const string x509ExtensionBase64String = "MCSCFW5vdC1yZWFsLXN1YmplY3QtbmFtZYILZXhhbXBsZS5jb20=";
+                byte[] x509ExtensionBytes = new byte[] {
+                    48,36,130,21,110,111,116,45,114,101,97,108,45,115,117,98,106,101,99,116,45,110,97,109,101,130,11,101,120,97,109,112,108,101,46,99,111,109
```

Comment:
Split to multi-line, and there should be a space after each number. E.g. `48, 36, 130....`